packages/ilib-loctool-mdx/MdxFileType.js

```diff
@@ -0,0 +1,300 @@
+/*
+ * MdxFileType.js - Represents a collection of Mdx files
+ *
+ * Copyright © 2025, Box, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var fs = require("fs");
+var path = require("path");
+var mm = require("micromatch");
+var MdxFile = require("./MdxFile.js");
+var YamlFileType = require('ilib-loctool-yaml');
+
+var MdxFileType = function(project) {
+    this.type = "md";
+    this.datatype = "mdx";
+    this.project = project;
+    this.API = project.getAPI();
+
+    this.logger = this.API.getLogger("loctool.lib.MdxFileType");
+
+    // default extensions
+    this.extensions = [ ".mdx", ".md" ];
+
+    this.extracted = this.API.newTranslationSet(project.getSourceLocale());
+    this.newres = this.API.newTranslationSet(project.getSourceLocale());
+    this.pseudo = this.API.newTranslationSet(project.getSourceLocale());
+
+    this.pseudos = {};
+
+    // generate all the pseudo bundles we'll need
+    project.settings && project.settings.locales && project.settings.locales.forEach(function(locale) {
+        var pseudo = this.API.getPseudoBundle(locale, this, project);
+        if (pseudo) {
+            this.pseudos[locale] = pseudo;
+        }
+    }.bind(this));
+
+    // for use with missing strings
+    if (!project.settings.nopseudo) {
+        this.missingPseudo = this.API.getPseudoBundle(project.pseudoLocale, this, project);
+    }
+
+    this.fileInfo = {
+        translated: [],
+        untranslated: []
+    }
+
+    this.yamlFileType = new YamlFileType(this.project);
+};
+
+/**
+ * Initialize the file type. This is called by the loctool's Project.init()
+ * method to allow file types to perform async initialization (e.g., loading
+ * ESM-only packages like remark-mdx).
+ *
+ * @param {Function} cb callback function to call when initialization is complete
+ */
+MdxFileType.prototype.init = function(cb) {
+    // Initialize the remark-mdx parser (ESM-only package)
+    MdxFile.initMdxParser(function(err) {
+        if (err) {
+            this.logger.error("Failed to initialize remark-mdx parser: " + err.message);
+            this.logger.error(err);
+            // Continue anyway - the error will be caught when parse() is called
+        }
+        cb();
+    }.bind(this));
```

Comment:
Matches suggested change to `initMdxParser` fn

```suggestion
    // Initialize the remark-mdx parser (ESM-only package)
    MdxFile.initMdxParser().then(function () {
        cb();
    }).catch(function (err) {
        this.logger.error("Failed to initialize remark-mdx parser: " + err.message);
        this.logger.error(err);
    }.bind(this));
```